Modules/StudyProgramme/classes/class.ilObjStudyProgrammeIndividualPlanGUI.php

```diff
@@ -83,21 +88,33 @@ public function __construct($a_parent_gui, $a_ref_id) {
 		$this->lng = $lng;
 		$this->user = $ilUser;
 		$this->assignment_object = null;
-		
+
 		$this->object = null;
 
 		$lng->loadLanguageModule("prg");
 
 		$this->tpl->addCss("Modules/StudyProgramme/templates/css/ilStudyProgramme.css");
 	}
-	
+
+	/**
+	* Get a (cached) instance of ilStudyProgrammeUserProgressDB
+	*
+	* @return ilStudyProgrammeUserProgressDB
+	*/
+	private function getStudyProgrammeUserProgressDB() {
+		if(! $this->sp_user_progress_db) {
+			 $this->sp_user_progress_db = ilObjStudyProgramme::getNewStudyProgrammeUserProgressDB();
+		}
+		return $this->sp_user_progress_db;
+	}
```

Comment:
We control the creation of `ilObjStudyProgrammeIndividualPlabGUI` so we could inject this dependency. I see no additional value in this getter, then, so you could use `$this->sp_user_progress_db` directly then.